### ISSUE TITLE
Technical and non-technical discussion without relying on proprietary github

### ISSUE THREAD

**Issue by zetok** [Tox/Tox-Foundation-Discussion#3]
There is a need for something that would allow to notify people about Tox related things, in a way that would allow them to provide technical feedback. It would be nice to have a real ML for that kind of stuff.

https://github.com/irungentoo/toxcore/pull/1148#issuecomment-65878431 ← too bad, no ML
https://github.com/irungentoo/toxcore/issues/1009#issue-39806722 ← too bad, no ML, you get instead some proprietary shit that is for non-technical stuff, there's no way to get some feedback from all people who would want to subscribe to ML, etc..

Allegedly there's ML for node maintainers, but from what I remember @TheLastProject had 0**(!)** mails received for few months. How exactly *telling* people that they don't have to maintain (update, etc.) their nodes at all benefits Tox?

Beside being ded, that ML isn't for discussion.

There are many other reasons for real ML, many of which were mentioned/listed/etc.

Also, please keep in mind that largest software project that exists (Linux kernel) uses mailing lists, since that's the only thing that allows developers to coordinate their efforts sufficiently. You may want to consider that when thinking about future of Tox.

Please change that state of things.

EDIT: ML = mailing list

Update
====
Mailing lists available at https://lists.tox.im

**Comment by tsudoko** [comment]
While we're at it, having non-github repository mirrors would be a good thing too.

**Comment by nurupo** [comment]
>Allegedly there's ML for node maintainers, but from what I remember @TheLastProject had 0(!) mails received for few months.

Updating nodes is important, but the main purpose of that mailing list is for when Tox gets a drastic change that would render the existing nodes useless. We had such changes few times before and we had to remove many nodes from the list just because we had no means of contacting their owners, thus we came up with the mailing list. That said, such changes don't happen too often. For example, I have been running a node built in summer 2014 until a week ago just fine. It's a good thing that that ML doesn't get emails often, it means that things don't break : )

>How exactly telling people that they don't have to maintain (update, etc.) their nodes at all benefits Tox?

No one is telling people that people shouldn't maintain their nodes, so please don't exaggerate.

**Comment by Dirius77** [comment]
A mailing list would be nice, as I personally like to look at issues, and see what people are discussing, and it's rather hard to have to check multiple places for that (All the different github issues across different Repos)

**Comment by urras** [comment]
What about support.libtoxcore.so?

**Comment by zetok** [comment]
@urras 

![broken support](https://cloud.githubusercontent.com/assets/3148759/5989811/63aca844-a98c-11e4-984c-415ba6205688.png)

Horribly broken without javascript on.
There is no way to just subscribe for notifications.
Account required.
No source.
In what way exactly it is better than github?

It's not even comparable to mailing list.

Aside from that, I personally didn't even remember about address of that "support", otherwise I would have made sure to highlight here that it can't cover all what is required…

**Comment by ProMcTagonist** [comment]
There exist a subreddit and a board on 8chan. Those are places that can be made use of more for discussion and opinion-gathering.

**Comment by zetok** [comment]
@ProMcTagonist 
Using some external board is a big fat no.

About subreddit, perhaps it can be used to gather some opinions. Still a lot of points I've highlighted earlier are valid in case of it. Mailing list is not so limited…

**Comment by stqism** [comment]
@Dirius77 Efforts to do centralised support systems and systems for sharing feedback never work because everyone comes back to github.

@zetok Normal people have Javascript enabled.

@ProMcTagonist The board isn't official in any way.

@tsudoko We've had those for over a year powered by Jenkins.

**Comment by stqism** [comment]
The thing about a mailing list is that 1, no one would use it; and 2, I doubt the majority of our users use them.

**Comment by zetok** [comment]
@stqism 

>Normal people have Javascript enabled.

I take that as a compliment :3

====

>1, no one would use it;

How do you know without trying?

>2, I doubt the majority of our users use them.

Yes. It won't be for majority of users. Please remember that one size doesn't fit all. When Tox is taking over market share, most of users may be content with what's already available (I doubt it…), but there will be some for whom existing stuff won't be enough. Even if that was just 1%, it still is a huge amount of people.

But lets come back to how things are. Some of current users already said that they want mailing list.
Please note that mailing lists, when it comes to software, tend to attract the kind of people who would answer ***Yes please*** to a question ***Would you like to run this horribly broken experimental software that may burn your house and kill your cat?***  It may be just my opinion, but I think that Tox needs and will need those people as users too, considering that they will provide useful feedback when it comes to how technically well software is doing.

Currently there isn't even a way to ask about something like that properly. This is important too, since (IIRC) not so long ago @irungentoo wanted some people to test some stuff, and hardly anyone answered. Mailing list would help with that aspect too. Feedback wouldn't necessarily be instantaneous, but it would be of good quality, and from people who would want to help.

**Comment by zetok** [comment]
@dcharles525
since you've asked in groupchat what is wrong with proprietary github…

Not so long ago there was a conversation about yet another reason why there should be mailing list.
Basically there was this guy who had bunch of accounts on github, and made some **useful** issues on toxcore repo. Github being github doesn't allow for multiple accounts.

They deleted all accounts.

At this point, I personally wouldn't care, if not for one thing…
**fucking github removed not only accounts, but also issues**

I only know about it because of dead links to those issues…
https://wiki.tox.im/Groupchat ← some of links here are dead, who knows how many other useful issues are irreversibly lost…


I hope that this provides at least some bit of insight why github cannot be trusted.

**Comment by louipc** [comment]
This is a wonderful idea.

**Comment by stqism** [comment]
@zetok 
It should be noted that they deleted all the issues and all those accounts because account fraud is against the Github terms of service.

**Comment by raumkraut** [comment]
FWIW, I agree that development discussion should ideally be hosted somewhere under the control of the Tox project itself. However, there are degrees of control (eg. owning hardware > renting server > VPS > 8chan > Github > Facebook), and it depends how direct the control really needs to be.

With that said; I despise mailing lists for anything more than broadcast announcements. IME, the interface is too unfriendly, esoteric and error-prone for users. I've found mailing lists to be a relatively high bar to casual involvement for anyone not an old-school UNIX type (eg. Linux kernel devs).

Plus, mailing lists tend to be all-or-nothing affairs, which is fine for someone heavily involved with the project, but for someone who just wants to ask one question, or report one bug, it can become a full-time job just trying to find the relevant responses.
Or then again, maybe that's just because modern email clients suck for using mailing lists. Yet most people would likely be using such a client, so I think the consideration remains.

But while we're considering technologies old enough to make a hipster blush; what about NNTP?

**Comment by zetok** [comment]
@raumkraut 

>I've found mailing lists to be a relatively high bar to casual involvement for anyone not an old-school UNIX type (eg. Linux kernel devs).

I'm afraid that you, as many others, are missing the point. As I've stated previously, there are already places for casual and a bit less casual users.
Mailing list would be something for those "old-school UNIX type" people, and other who prefer to use mailing list.


>But while we're considering technologies old enough to make a hipster blush; what about NNTP?

Email is something **everyone** uses. What kind of benefit do you see in using something that practically no one uses and no one knows about in modern times?

**Comment by louipc** [comment]
On Tue 03 Feb 2015 06:08 -0800, raumkraut wrote:
> FWIW, I agree that development discussion should ideally be hosted
> somewhere under the control of the Tox project itself. However, there
> are degrees of control (eg. owning hardware > renting server > VPS >
> 8chan > Github > Facebook), and it depends how direct the control
> really needs to be.

> With that said; I despise mailing lists for anything more than
> broadcast announcements. IME, the interface is too unfriendly,
> esoteric and error-prone for users. I've found mailing lists to be a
> relatively high bar to casual involvement for anyone not an old-school
> UNIX type (eg. Linux kernel devs).

Wow. Must be some cultural disconnect or something. I find email much
more convenient than web-based communication. I can read and respond
offline. I get a plethora of interface options from console ncurses to
gtk/qt gui, to webmail clients to mobile -for free-. If you're using an
error-prone interface, then change your interface.

> Plus, mailing lists tend to be all-or-nothing affairs, which is fine
> for someone heavily involved with the project, but for someone who
> just wants to ask one question, or report one bug, it can become a
> full-time job just trying to find the relevant responses.  Or then
> again, maybe that's just because modern email clients suck for using
> mailing lists. Yet most people would likely be using such a client, so
> I think the consideration remains.

If you're doing email properly, responses are conveniently threaded.
 
> But while we're considering technologies old enough to make a hipster
> blush; what about NNTP?

I'm actually responding to this thread via email in a terminal emulator.
Github had the good sense to implement decent email support. What.

The point is if you like it you will use it. If you don't like it, then
don't use it. There are still a lot of people that know how to use email
and prefer it.

Learn how to use email.
Leave the tweeting to the twits.

**Comment by Dirius77** [comment]
Actually I've been responding to every single one of these from my email
(By 'these' I mean the Tox issues I discuss in github). So I think using a
mailing list would be a very good way for this to work.

**Comment by Timodz** [comment]
Go ahead guys its a nice app and will be nicer if those features have been added :) +1:

**Comment by draziw-** [comment]
This is actually a great idea.  Why we just don't make it ?
We can talk technical using mailing lists, we can talk issues in github issues and that's great.

**Comment by ProMcTagonist** [comment]
Stqism is creating a trial mailing list. If it's neglected it will be killed after a month, and "If people use it it'll stay".

**Comment by Dirius77** [comment]
How do we get added to this mailing list.

**Comment by stqism** [comment]
Still working on figuring it out, GNU Mailman has got to be the worst software ever.

**Comment by optimumtact** [comment]
come on stq, please

**Comment by zetok** [comment]
Update: 
====
@stqism lent a vps to set up ML on it.

I've tried my best to set it up right, but sadly due to stq's setup I am not even able to test it.


Basically there was happy http redirection of ``lists.tox.im`` to vps' IP. Then stq went route ``muh no http hurr durr`` and now there's https only. *Obviously* there are no certs supplied, and "testing" with only IP doesn't work, since mailing lists for ``lists.tox.im`` aren't even displayed for just IP.

Before he cut off http, I've noticed at least one issue that needed to be fixed for mailing lists (probably nginx misconfiguration on my part), but I can't test for it, since mailing lists aren't displayed when going to the website by IP.

```
[20:26:11] <NikolaiToryzin> (…) I have to go in and verify everything then disable external access
[20:26:27] <NikolaiToryzin> Before we can put some certs on it and get nginx working
```

↑ That's the status since some time already.

**Comment by zetok** [comment]
Update2:
====

@stqism said to just use self-signed certs. Too bad that there are problems with them that make it hard for people to actually access and  test web interface.

Aside from that, with fake cert I managed to fixed problem, and sending email through ML to myself worked just fine.

Thus the only part is waiting for stq to do his part.

**Comment by redsteakraw** [comment]
As far as Git is concerned other projects use other services for example Kodi(XBMC) and Qt both use gitorious, [KDE host's their own git web frontend](https://projects.kde.org) and some stick to GitHub.  As for Bug tracking projects use bugzilla but again [KDE has a nice web interface](https://bugs.kde.org/ )Mailing lists are the standard along with IRC channels. However I feel for Tox to progress Devs should dogfood and use tox groupchats instead of IRC.

**Comment by dubslow** [comment]
I would like to ditch IRC once the new groupchats are done, yes.

**Comment by stqism** [comment]
@dubslow Lack of mobile support and history syncing makes that a non-option.

Of course we'll see when they are finished.


@zetok "Before he cut off HTTP" 
Please don't lie, I explained to you how HSTS headers work and how we've had one for over a year.

Regardless, I have to go in and vet everything fully before I can toss on a proper cert.

**Comment by zetok** [comment]
@stqism 
>Please don't lie, I explained to you how HSTS headers work and how we've had one for over a year.

That's how it looked for days from my side, before you finally said why it actually doesn't work, *coincidentally* just after my post here…

If you don't want things to escalate, please work on your communication, instead of just saying “Tox doesn't support http”.


Aside from that… since ML is finally up, I've asked @stqism to make those lists: https://gist.github.com/zetok/fd598d7df35f49d218ab

*I am really curious which ones will actually be there*

**Comment by stqism** [comment]
@zetok I disagree with your list, but General and Development seemed like good starts for a 0 volume list.

**Comment by stqism** [comment]
As the ml is up and https://git.tox.im is up I'd call this resolved.
